tools/interactive_fit.py

```diff
@@ -0,0 +1,359 @@
+        self.fig.subplots_adjust(bottom=bot)
+
+        xpoints = []
+        for x_min, x_max, n_point in zip(x_mins, x_maxs, n_points):
+            xpoints.append(np.linspace(x_min, x_max, n_point))
+        self.projections = list(combinations(range(self.nvars), 2))
+
+        f = interpolator(self.xdata.T, self.ydata)
+        def master_interpolator(**kwargs):
+            """Converts keyword arguments to positional arguments,
+            and applies it to the interpolator"""
+            args = [0]*len(kwargs)
+            for i, v in enumerate(self.vars):
+                args[i] = kwargs[v.name]
+            return f(*args)
+
+        ncols = np.ceil(len(self.projections)**0.5)
+        nrows = ncols
+
+        self.plots = {}
+        plotnr = 1
+        for x1, x2 in self.projections:
+            ax = self.fig.add_subplot(ncols, nrows, plotnr, projection="3d")
+            ax.set_axis_off()  # Remove back panes.
+
+            xydata = np.array([self.xdata[x1], self.xdata[x2]])
+            mesh = np.meshgrid(xpoints[x1], xpoints[x2])
+            
+            # Pre-apply all the data that will not change for this projection
+            z_function = functools.partial(self.model,
+                                           **{self.vars[x1].name: mesh[0],
+                                              self.vars[x2].name: mesh[1]})
+            y_interpolator = functools.partial(master_interpolator,
+                                               **{self.vars[x].name: self.xdata[x] for x in (x1, x2)})
+
+            title = "{}{}".format(*[self.vars[x].name for x in (x1, x2)])
+
+            p = ProjectionPlot(ax, xydata, y_interpolator, mesh, z_function, title)
+            self.plots[(x1, x2)] = p
+            plotnr += 1
+
+        self._draw_sliders()
+        self._update_plot(self.params[0])  # Update all plots.
+        plt.show()
+
+
+if __name__ == "__main__":
+    def f(xy, a, b):
+        x, y, a = xy
+        return np.cos(a*x) * b*np.sin(y)
+
+    from symfit.api import Parameter, Variable, sin, cos
+
+    def f_sym(x, y, a, b):
+        return cos(a*x) * b*sin(y)
+
+    xdata = np.linspace(-np.pi, np.pi, 7)
+    ydata = np.linspace(-np.pi, np.pi, 7)
+    adata = np.linspace(0, 2.5, 7)
+
+    xx, yy, aa = np.meshgrid(xdata, ydata, adata)
+
+    xydata = np.array((xx.flatten(), yy.flatten(), aa.flatten()))
+    #xydata = np.array((xx.flatten(), yy.flatten()))
+    zs = np.array([f(xy, 1.5,  2) for xy in xydata.T])
+    zdata = zs.reshape(xx.shape)
+#    print(xydata.T.shape, zdata.flatten().shape)
+#    interp = NearestNDInterpolator(xydata.T, zdata.flatten())
+#    print(interp(0.5, 1, 0.2))
+#    interp2 = LinearNDInterpolator(xydata.T, zdata.flatten())
+#    print(interp2(0.5, 1, 0.2))
+
+    x = Variable()
+    y = Variable()
+    a = Variable()
+    b = Parameter()
+
+    model = cos(a*x) * b*sin(y)
+    model = f_sym(x, y, a, b)
+    fit = InteractiveFit3D(model, xydata.T, zdata.flatten())
```

Comment:
So if I did everything correctly, this should now be simply
```python
    InteractiveFit3D(model, xx, yy, zdata)
```
Although I would prefer even in such a simple case to use keyword arguments for readability:
```python
    InteractiveFit3D(model, x=xx, y=yy, z=zdata)
```